integration/update_field_compat_test.go

```diff
@@ -54,6 +56,97 @@ func TestUpdateFieldCompatInc(t *testing.T) {
 	testUpdateCompat(t, testCases)
 }
 
+func TestUpdateFieldCompatMax(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]updateCompatTestCase{
+		"Int32Lower": {
+			update:        bson.D{{"$max", bson.D{{"v", int32(30)}}}},
+			skipForTigris: "fixing tigris validation needed",
+		},
+		"Int32Higher": {
+			update:        bson.D{{"$max", bson.D{{"v", int32(60)}}}},
+			skipForTigris: "fixing tigris validation needed",
+		},
+		"Int32Negative": {
+			update:        bson.D{{"$max", bson.D{{"v", int32(-22)}}}},
+			skipForTigris: "fixing tigris validation needed",
+		},
+		"Document": {
+			update: bson.D{{"$max", bson.D{{"v", bson.D{{"foo", "bar"}}}}}},
+			skip:   "https://github.com/FerretDB/FerretDB/issues/457",
+		},
+		"EmptyDocument": {
+			update: bson.D{{"$max", bson.D{{"v", bson.D{{}}}}}},
+			skip:   "https://github.com/FerretDB/FerretDB/issues/1000",
+		},
+		"Double": {
+			update: bson.D{{"$max", bson.D{{"v", 54.32}}}},
+		},
+		"DoubleNegative": {
+			update:        bson.D{{"$max", bson.D{{"v", -54.32}}}},
+			skipForTigris: "fixing tigris validation needed",
+		},
+		"NotExisting": {
+			update:        bson.D{{"$max", bson.D{{"v", int32(60)}}}},
+			skipForTigris: "fixing tigris validation needed",
+		},
+
+		"MultipleQueries": {
+			update:        bson.D{{"$max", bson.D{{"v", int32(39)}, {"a", int32(30)}}}},
+			skipForTigris: "fixing tigris validation needed",
+		},
+		"DuplicateQuery": {
+			update: bson.D{{"$max", bson.D{{"v", int32(39)}, {"v", int32(30)}}}},
+			skip:   "Handle duplicates correctly",
+		},
+
+		// Strings are not converted to numbers
+		"StringIntegerHigher": {
+			update: bson.D{{"$max", bson.D{{"v", "60"}}}},
+		},
+		"StringIntegerLower": {
+			update: bson.D{{"$max", bson.D{{"v", "30"}}}},
+		},
+		"StringDouble": {
+			update: bson.D{{"$max", bson.D{{"v", "54.32"}}}},
+		},
+		"StringDoubleNegative": {
+			update: bson.D{{"$max", bson.D{{"v", "-54.32"}}}},
+		},
+		"StringLexicographicHigher": {
+			update: bson.D{{"$max", bson.D{{"v", "goo"}}}},
+		},
+		"StringLexicographicLower": {
+			update: bson.D{{"$max", bson.D{{"v", "eoo"}}}},
+		},
+		"StringLexicographicUpperCase": {
+			update: bson.D{{"$max", bson.D{{"v", "Foo"}}}},
+		},
+		"BoolTrue": {
+			update: bson.D{{"$max", bson.D{{"v", true}}}},
+		},
+		"BoolFalse": {
+			update:        bson.D{{"$max", bson.D{{"v", false}}}},
+			skipForTigris: "fixing tigris validation needed",
+		},
+		"EmptyOperand": {
+			update:     bson.D{{"$max", bson.D{}}},
+			resultType: emptyResult,
+		},
+		"DateTime": {
+			update:        bson.D{{"$max", bson.D{{"v", primitive.NewDateTimeFromTime(time.Date(2021, 11, 1, 12, 18, 42, 123000000, time.UTC))}}}},
+			skipForTigris: "fixing tigris validation needed",
```

Comment:
I don't think it could be fixed at all :) 

So, if you want this test to work for Tigris, you must work with a data set where v is defined as a date. Tigris doesn't allow to store multiple types in the same field. 

A possible solution would be to separate those test cases into a few different functions based on data (e.g. one function for bools, another function for doubles, another function for date etc). Each function will call `setup.Setup` for a particular data provider (separate for bools, separate for ints etc). This way it will work for Tigris.

However, such test cases will limit the number of situations we check for Postgres (e.g. for postgres we might want to set a date to the field that stored bools). I believe this should be a separate test case for postgres only (where we  can use "skip for Tigris" and set a particular reason that Tigris doesn't support such data types). 

Let's discuss if everyone agrees with this approach.

Reply:
> So, if you want this test to work for Tigris, you must work with a data set where v is defined as a date.

Correct. We have that dataset (`DateTimes`).

> Tigris doesn't allow to store multiple types in the same field.

That's also correct.

> A possible solution

Sorry, but for what problem?

Reply:
> Sorry, but for what problem?

For the case when we want to check that storing various data types in the same field works correctly for Postgres.

Reply:
@AlekSi or I don't understand your question. This is why sometimes I prefer synchronous calls over such discussions.

Reply:
If I understand correctly you wan to move these test cases to integration tests?

Reply:
That should be handled by `testUpdateCompat`. In fact, there is already a code for that:

https://github.com/noisersup/FerretDB/blob/35b050fd7ce29614d018b2f5fd091e546eb6db42/integration/update_compat_test.go#L128-L133

It might not work, but then we should just™ fix it

Reply:
So, to resume:
- if we want to test it with Tigris, we need to split those test cases into groups based on particular data types
- if after that we want to test that storing various data types still works for postgres, we need to add an additional postgres-only test for it.

Reply:
@AlekSi the point is that tigris doesn't return any error in that case, because if it's possible they parse your input to the current existing type

Reply:
@noisersup as agreed at the daily sync, please mention this issue https://github.com/FerretDB/FerretDB/issues/1061 in "skipForTigris", and that will be it for this task.